middle_end/flambda/inlining/inlining_decision.ml

```diff
@@ -49,14 +49,19 @@ module Function_declaration_decision = struct
         large_function_size: Code_size.t;
       }
 
-  let can_inline t =
+  type result =
+    | Not_inlinable
+    | Always_inlinable
+    | Maybe_inlinable
+
+  let to_result t =
```

Comment:
The name `result` is really a bit too broad, maybe we could rename the type, and this function to something like `inlining_status` or `inlining_behavior` ?

Reply:
"summary", perhaps?

Reply:
I decided to go with inlining_behavior